client/src/partials/inventory/list/modals/actions.modal.js

```diff
@@ -0,0 +1,136 @@
+angular.module('bhima.controllers')
+.controller('InventoryListActionsModalController', InventoryListActionsModalController);
+
+InventoryListActionsModalController.$inject = [
+  'AccountService', 'InventoryService', 'InventoryGroupService',
+  'InventoryTypeService', 'InventoryUnitService', 'NotifyService',
+  '$uibModalInstance', 'Store', 'data'
+];
+
+function InventoryListActionsModalController(Account, Inventory, InventoryGroup, InventoryType, InventoryUnit, Notify, Instance, Store, Data) {
+  var vm = this, session = vm.session = {};
+
+  // variables for storage
+  var GroupStore = {},
+      TypeStore  = {},
+      UnitStore  = {};
+
+  // map for actions
+  var map = { 'add' : addList, 'edit' : editList };
+
+  // expose to the view
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  // startup
+  startup();
+
+  /** submit data */
+  function submit(form) {
+    if (form.$invalid) {
+      return;
+    }
+
+    var record = cleanForSubmit(vm.session);
+    map[vm.action](record, vm.identifier)
+    .then(function (res) {
+      Instance.close(res);
+    });
+  }
+
+  /** add inventory list */
+  function addList(record) {
+    return Inventory.create(record)
+    .then(function (res) {
+      return res;
+    })
+    .catch(Notify.errorHandler);
+  }
+
+  /** edit inventory list */
+  function editList(record, uuid) {
+    return Inventory.update(uuid, record)
+    .then(function (res) {
+      return res;
+    })
+    .catch(Notify.errorHandler);
+  }
+
+  /** cancel action */
+  function cancel() {
+    Instance.dismiss();
+  }
+
+  /** format data to data structure in the db */
+  function cleanForSubmit(session) {
```

Comment:
This should really be in a service, probably the `InventoryService`.

@sfount, would your [util.filterDirtyFormElements()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/util.js#L13) function help out here?